content/docs/command-reference/exp/branch.md

```diff
@@ -0,0 +1,43 @@
+# exp branch
+
+Commit the results from an [experiment](/doc/command-reference/exp) in a new Git
+branch, which will become the <abbr>workspace</abbr>.
+
+## Synopsis
+
+```usage
+usage: dvc exp branch [-h] [-q | -v] experiment branch
+
+positional arguments:
+  experiment     Experiment to be promoted
+  branch         Git branch name to use
+```
+
+## Description
+
+Makes a
+[Git branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
+off the last commit (`HEAD`) based on the given `experiment`, using the `branch`
+name provided. The new branch is switched into (`git checkout`).
+
+In most cases this is equivalent to using `dvc exp apply` (applies the
+`experiment` results to the workspace) followed by Git branching and committing:
+
+```dvc
+$ dvc exp apply experiment
+$ git checkout -b branch
+$ git add . && git commit
+```
+
+For [checkpoints](/doc/command-reference/exp/run#checkpoints), the `experiment`
+(custom Git branch) is merged into the new `branch`.
```

Comment:
If I understand correctly, `exp branch` is equivalent to `git branch <exp_name> <rev>`. This again seems like additional work to explain in other terms what's actually happening. If users understand the experiments already have their own commits (multiple in the cases of checkpoints), it seems easier to understand this command.

Reply:
This is correct. There's multiple ways you could represent what `exp branch` does using git examples:

```
$ git branch <exp_name> <rev>
```
```
$ git branch <exp_name>
$ git checkout <exp_name>
$ git merge <exp_rev>
$ git checkout <original HEAD>
```
```
$ git checkout -b <exp_name>
$ git merge <exp_rev>
$ git checkout <original HEAD>
```

All of these do exactly the same thing as far as git is concerned. The first example is the simplest, but I'm not sure if that's a git workflow that typical DVC users would recognize without having to go look up the docs for `git branch` to see what it does.

Reply:
> If I understand correctly, exp branch is equivalent to `git branch <exp_name> <rev>`

I think that's only true when your `HEAD` is also the base of the given `experiment`, which may be the most common scenario but not always.

Reply:
> I think that's only true when your `HEAD` is also the base of the given `experiment` though? That may be the most common scenario, but not always the case.

For at least the first two commands, it shouldn't matter what your HEAD is, since they do not checkout anything from your current HEAD. For the last two, they would need a first step like `git checkout <base_rev>` if the current HEAD diverged from the experiment.

Also, I forgot that `exp branch` requires a branch name as an argument, so according to the docs, `<exp_name>` should really be `<branch>` in all these examples. Maybe worth considering whether the branch name arg should be optional, especially since users already have the option to name the experiments when creating them, but that's a note for another time.

To get back to the main point, I don't think we should keep:

```
$ dvc exp apply experiment
$ git checkout -b branch
$ git add . && git commit
```

It's inaccurate since `exp branch` is neither applying an experiment (I know this was an earlier confusion that has since been resolved) nor committing anything. My suggestion would be remove this. If we need more description, how about something like:

```
A new branch is created with the history of that experiment.
For [checkpoints](/doc/command-reference/exp/run#checkpoints), 
each checkpoint will have its own commit in the new branch.
```

Reply:
> For the last two, they would need a first step like git checkout <base_rev> 

This is the part I didn't think was happening (maybe @pmrowla can confirm). I should double check all this anyway ⌛

> worth considering whether the branch name arg should be optional, especially since users already have the option to name the experiments when creating them, but that's a note for another

- Will note that somewhere...

> I don't think we should keep... It's inaccurate

We aren't saying that's exactly how `exp branch` is implemented, we say that it's "Equivalent to" that. Unless I'm (also) getting `exp apply` wrong, that seems correct i.e. the result is the same (BTW those sample commands have since been updated though).